### PR TITLE
`every_n_val_epochs` -> `every_n_epochs`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Deprecate `every_n_val_epochs` in favor of `every_n_epochs`

The flag is used in the `on_validation_end` hook, but we will also want to use it in the `on_train_epoch_end` hook

Part of #7724

### Does your PR introduce any breaking changes ? If yes, please list them.

None

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [x] Did you list all the breaking changes introduced by this pull request?

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -108,26 +108,33 @@ class ModelCheckpoint(Callback):
...
         train_time_interval: Checkpoints are monitored at the specified time interval.
             For all practical purposes, this cannot be smaller than the amount
             of time it takes to process a single training batch. This is not
             guaranteed to execute at the exact time specified, but should be close.
-            This must be mutually exclusive with ``every_n_train_steps`` and ``every_n_val_epochs``.
-        every_n_val_epochs: Number of validation epochs between checkpoints.
-            If ``every_n_val_epochs == None or every_n_val_epochs == 0``, we skip saving on validation end.
-            To disable, set ``every_n_val_epochs = 0``. This value must be ``None`` or non-negative.
+            This must be mutually exclusive with ``every_n_train_steps`` and ``every_n_epochs``.
+        every_n_epochs: Number of epochs between checkpoints.
```
Comment (awaelchli) [inline pytorch_lightning/callbacks/model_checkpoint.py:117]:
```suggestion
        every_n_epochs: Number of training epochs between checkpoints.
```
not to be ambiguous, but I guess it is already clear we mean training epochs

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/callbacks/model_checkpoint.py:117]: The rename in this PR is so that it can be used for validation epochs and training epochs.

So we cannot specify here

Comment (ananthsub) [reply pytorch_lightning/callbacks/model_checkpoint.py:117]: @carmocca why is every_n_epochs along with a boolean flag to save on train epoch end preferable to supporting two flags like every_n_val_epochs and every_n_train_epochs?

The latter allows us to adjust the checkpoint frequency separately for training and validation. The mutual exclusion check is something we could address in the callback implementation

**Comment by ananthsub** [comment]
@carmocca why is every_n_epochs along with a boolean flag to save on train epoch end preferable to supporting two flags like every_n_val_epochs and every_n_train_epochs?

The latter allows us to adjust the checkpoint frequency separately for training and validation. The mutual exclusion check is something we could address in the callback implementation.

**Comment by carmocca** [comment]
@ananthsub Is your question in the context that the user can set both `save_on_train_epoch_end` and `save_on_validation_end`? The current design only has the former which acts as a exclusive boolean already. So it is natural that the `every_n_epochs` flags follows the same pattern.

If there is a need to specify both, I think two ModelCheckpoint instances should be created. One with each boolean value, but not different flags.

Also this is seems better to avoid user confusion about the flag differnce

**Comment by ananthsub** [comment]
@carmocca it's more that as a user, users need to now know about the default behavior of `save_on_train_epoch_end` which is less explicit
- When reading the code, there's ambiguity around `ModelCheckpoint(every_n_epochs=...)`. Users need to check the default value for `save_on_train_epoch_end` to know when exactly end of epoch checkpoints are being saved
- Same goes for writing the code: Users need to set either `ModelCheckpoint(every_n_epochs=...) or `ModelCheckpoint(every_n_epochs=..., save_on_train_epoch_end)` which is less consistent than `ModelCheckpoint(every_n_train_epochs=...)` or `ModelCheckpoint(every_n_val_epochs=...)`

**Comment by carmocca** [comment]
> there's ambiguity around ModelCheckpoint(every_n_epochs=...). Users need to check the default value for save_on_train_epoch_end to know when exactly end of epoch checkpoints are being saved

You could also make a similar argument that the user would need to know if `save_on_train_epoch_end=True` is set with `every_n_train_epochs=...` or `save_on_train_epoch_end=False` is set with `every_n_val_epochs=...` 

> which is less consistent thanModelCheckpoint(every_n_train_epochs=...)orModelCheckpoint(every_n_val_epochs=...)`

Are you saying `every_n_train_epochs` would implicitly set `save_in_train_epoch_end=True` and the opposite for `every_n_val_epochs`?
